src/analytics/lambdas/user-segments-workflow/segment-job-status.ts

```diff
@@ -64,20 +74,95 @@ const _handler = async (event: SegmentJobStatusEvent) => {
         ReturnValues: 'ALL_NEW',
       });
 
-      // TODO: retrieve segment result including sample data from S3, update result to job status table
-
       logger.info(`Update segment job status to ${jobStatus} in DDB metadata table`);
       await ddbDocClient.send(command);
+    } else if (jobStatus === SegmentJobStatus.COMPLETED) {
+      const s3Path = `${process.env.SEGMENTS_S3_PREFIX}app/${appId}/segment/${segmentId}/job/${jobRunId}/`;
+
+      // Retrieve segment result including sample data from S3, update job status table
+      const summaryData = await s3Client.send(new GetObjectCommand({
+        Bucket: process.env.PIPELINE_S3_BUCKET,
+        Key: `${s3Path}segment-summary_000.csv`,
```

Comment:
Why is the file name fixed? Should we extract it as a constant?

Reply:
The segment job will output two csv files, one is named *segment_000.csv* for segment full data, the other is *segment-summary_000.csv* for segment summary info. We can define the name prefix and Redshift UNLOAD output will append partition number from `000`. We set `PARALLEL OFF`, so there is only one output file. I will extract the file name as a constant.